src/core/thread/address_resolver.cpp

```diff
@@ -281,7 +281,11 @@ void AddressResolver::HandleAddressNotification(Coap::Header &aHeader, Message &
             mCache[i].mTimeout = 0;
             mCache[i].mFailures = 0;
             mCache[i].mState = Cache::kStateCached;
-            SendAddressNotificationResponse(aHeader, aMessageInfo);
+
+            SuccessOrExit(mCoapServer.SendEmptyResponse(aHeader, aMessageInfo));
```

Comment:
I think we still want to call `HandleResolved()` below, even if we failed to send a response.